lynx/common/scene.py

```diff
@@ -10,28 +11,42 @@
 @dataclass
 class Scene(Serializable):
     entities: List[Entity] = field(default_factory=list)
-    _object_position_map: Dict[Vector, Object] = field(default_factory=dict)
+    _square_position_map: Dict[Vector, Square] = field(default_factory=dict)
     _object_id_map: Dict[int, Object] = field(default_factory=dict)
 
+    def _get_square(self, position: Vector) -> Square:
+        # This method guarantees that the square is correct
+        if square := self._square_position_map.get(position):
+            return square
+        else:
+            new_square = Square()
+            self._square_position_map[position] = new_square
+            return new_square
+
+    def _add_to_maps(self, entity: Entity) -> NoReturn:
+        if type(entity) is Object:
+            self._get_square(entity.position).append(entity)
+            self._object_id_map[entity.id] = entity
+
     def post_populate(self) -> NoReturn:
         for entity in self.entities:
-            if type(entity) is Object:
-                self._object_position_map[entity.position] = entity
-                self._object_id_map[entity.id] = entity
+            self._add_to_maps(entity)
 
     def add_entity(self, entity: Entity) -> NoReturn:
         self.entities.append(entity)
-        if type(entity) is Object:
-            self._object_position_map[entity.position] = entity
-            self._object_id_map[entity.id] = entity
+        self._add_to_maps(entity)
 
     def get_object_by_id(self, id: int) -> Union[Object, None]:
         return self._object_id_map.get(id)
-    
-    def get_object_by_position(self, position: Vector) -> Union[Object, None]:
-        return self._object_map.get(position)
-    
+
+    def get_objects_by_position(self, position: Vector) -> Union[List[Object], None]:
+        if square := self._square_position_map.get(position):
+            return square.objects
+        else:
```

Comment:
Maybe replace it with (because the earlier return is usedt):
```
   def get_objects_by_position(self, position: Vector) -> Union[List[Object], None]:
        if square := self._square_position_map.get(position):
            return square.objects
      
        self._square_position_map
        return []
```

Reply:
Right but haven't you noticed the extremely useful code `self._square_position_map` :D. Forgot to use `_get_square` method which now makes the whole function go: `self._get_square(position).objects`